.github/workflows/tests-on-push.yml

```diff
@@ -10,70 +10,145 @@ on:
+    name: Lint code
+    runs-on: ubuntu-latest
+    steps:
+      # Setup
+      - name: Checkout source
+        uses: actions/checkout@v3
+        with:
+          submodules: recursive
+      - name: Install poetry
+        run: |
+          curl -sSL https://install.python-poetry.org | python3 -
+          poetry self add poetry-exec-plugin
+      - name: Set up Python 3.11
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.11
+          # attempt to restore dependencies from cache (if successful, the following `poetry install` will be a no-op)
+          cache: poetry
+      - name: Install Python dependencies
+        run: poetry install
+      # run tests
       - name: Linting with Pylint
         run: poetry run pylint src test
-
       - name: Linting with mypy
         run: poetry run mypy .
-
       - name: Linting with black
         run: poetry run black --check .
-
       - name: Linting with isort
         run: poetry run isort . --check --diff
-
       - name: Linting with darglint
         run: poetry exec darglint
 
-
-      # Run unittests
-      ###############
+  unittests:
+    name: Run Unittests
+    runs-on: ubuntu-latest
+    steps:
+      # Setup
+      - name: Checkout source
+        uses: actions/checkout@v3
+        with:
+          submodules: recursive
+      - name: Install poetry
+        run: |
+          curl -sSL https://install.python-poetry.org | python3 -
+          poetry self add poetry-exec-plugin
+      - name: Set up Python 3.11
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.11
+          # attempt to restore dependencies from cache (if successful, the following `poetry install` will be a no-op)
+          cache: poetry
+      - name: Install Python dependencies
+        run: poetry install
+      # run tests
       - name: unittests
         run: poetry run pytest test/unittests/
       
+  e2e-tests:
+    name: Run E2E/Integration Tests
+    runs-on: ubuntu-latest
+    steps:
+      # Setup
+      - name: Checkout source
+        uses: actions/checkout@v3
+        with:
+          submodules: recursive
+      - name: Install poetry
+        run: |
+          curl -sSL https://install.python-poetry.org | python3 -
+          poetry self add poetry-exec-plugin
+      - name: Set up Python 3.11
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.11
+          # attempt to restore dependencies from cache (if successful, the following `poetry install` will be a no-op)
+          cache: poetry
+      - name: Install Python dependencies
+        run: poetry install
+      # run tests
+      - name: start stack
+        run: poetry run dsp-tools start-stack --no-prune
+      - name: e2e tests
+        run: poetry run pytest test/e2e/
+      - name: stop stack  # see if this command can run (it isn't tested anywhere else)
+        run: poetry run dsp-tools stop-stack
+
+  others:
+    name: Other Tests
```

Comment:
Look out: these are no tests, that's the setup that is required for one of the e2e tests (for the local image processing (fast xmlupload)). 
That's why the e2 tests failed.

Reply:
thanks for the hint, clearly I wasn't paying attention

Reply:
> On line 127, we could tell pytest to execute all except `test/e2e/test_fast_xmlupload.py`. And then, you can add a job on line 171 that runs only that one.

I simply put this in front of all of them. Good enough?

Reply:
sure